File: JavaScript/Semana 3/Bubble.js
function bubbleSort(array) {
    let s = array.length;
    let swapped;
    for(let i = 0; i < s - 1; i++){
        swapped = false;
        for(let j = 0; j < s - i - 1; j++){
            if(array[j] > array[j + 1]){
                let temp = array[j];
                array[j] = array[j + 1];
                array[j + 1] = temp;
                swapped = true;
            }
        }
        if(!swapped){
            break;
        }
    }
}

function main(){
    let array = [7,23,5,235,4,7];
    console.log("Antes de ordenar los elementos del array: ");
    console.log(array);

    bubbleSort(array);

    console.log("Después de ordenar los elementos del array: ");
    console.log(array);

}

main();

File: Java/Semana 3/Bubble.java
public class Bubble {

    public static void bubbleSort(int[] array) {
        int s = array.length;
        boolean swapped;
        for(int i = 0; i < s - 1; i++){
            swapped = false;
            for(int j = 0; j < s - i - 1; j++){
                if(array[j] > array[j + 1]){
                    int temp = array[j];
                    array[j] = array[j + 1];
                    array[j + 1] = temp;
                    swapped = true;
                }
            }
            if(!swapped){
                break;
            }
        }
    }

    public static void main(String[] args) {
        int[] array = {5, 3, 8, 4, 2};
        System.out.println("Antes de ordenar los elementos del array: ");
        for(int i = 0; i < array.length; i++){
            System.out.print(array[i] + " ");
        }

        bubbleSort(array);
        
        System.out.println("\nDespues de ordenar los elementos del array: ");
        for(int i = 0; i < array.length; i++){
            System.out.print(array[i] + " ");
        }
    }

}